<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME)" />

  <arg name="standalone" default="true" />
  <arg name="debug" default="false" />
  <arg name="custom_config" default="" />
  <arg name="node_name" default="freespace_pc" />
  <arg name="name_suffix" default="" />
  <arg name="topic_3d_lidar_in" default="os_cloud_nodelet/points" />
  <arg name="topic_3d_lidar_out" default="~points_freespace" />

  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <!-- Standalone -->
  <arg if="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="standalone" />
  <arg if="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="" />

  <!-- Nodeleted -->
  <arg name="nodelet_manager_name" default="$(arg UAV_NAME)_pcl_nodelet_manager" />
  <arg if="$(eval not arg('standalone') and not arg('debug'))" name="nodelet" value="load" />
  <arg if="$(eval not arg('standalone') and not arg('debug'))" name="nodelet_manager" value="$(arg nodelet_manager_name)" />

  <arg if="$(eval arg('name_suffix') == '')" name="nodelet_name" value="$(arg node_name)" />
  <arg unless="$(eval arg('name_suffix') == '')" name="nodelet_name" value="$(arg node_name)_$(arg name_suffix)" />

  <group ns="$(arg UAV_NAME)">

    <node pkg="nodelet" type="nodelet" name="$(arg nodelet_name)" args="$(arg nodelet) mrs_pcl_tools/FreespacePointcloudCreator $(arg nodelet_manager)" output="screen" launch-prefix="$(arg launch_prefix)">

      <!-- Parameters -->
      <param name="uav_name" type="string" value="$(arg UAV_NAME)" />

      <rosparam file="$(find mrs_pcl_tools)/config/freespace_pc.yaml" />
      <rosparam if="$(eval not arg('custom_config') == '')" file="$(arg custom_config)" command="load" />

      <!-- Subscribers -->
      <remap from="~pointcloud_in" to="$(arg topic_3d_lidar_in)" />

      <!-- Publishers -->
      <remap from="~freespace_pointcloud_out" to="$(arg topic_3d_lidar_out)" />

    </node>

  </group>

</launch>
